### PR TITLE
Fix tests

### LINKED ISSUE #419: Unit tests failing with `IncompleteRead`
See the latest CI runs: https://github.com/devopshq/artifactory/actions

Errors include:
```
requests.exceptions.ChunkedEncodingError: ('Connection broken: IncompleteRead(236 bytes read, -236 more expected)', IncompleteRead(236 bytes read, -236 more expected))
urllib3.exceptions.IncompleteRead: IncompleteRead(236 bytes read, -236 more expected)
```

Should be fixed by https://github.com/getsentry/responses/pull/643 (thanks @beliaev-maksim!) but I wanted to open an issue to track that because:
- I spent a bit of time today trying to track it down and possibly fix it
- Assuming the above is the fix, we're waiting on a release on `responses`, and then we should update the dev requirements to set that version minimum probably: https://github.com/devopshq/artifactory/blob/master/requirements-dev.txt#L9

### PR DESCRIPTION
**Pull request by beliaev-maksim**
* fix the way how we install dependencies. It should be done via extras
* updated status code in tests
* bumped requests rependency

closes #419

### CODE REVIEW

**Review by briantist — COMMENTED** [review]
This looks really good to me, looking forward to having the tests working again.
Could you also update the contributing doc with the details about how to install dev requirements?
https://github.com/devopshq/artifactory/blob/master/docs/CONTRIBUTE.md

File: docs/CONTRIBUTE.md [hunk 1]
```diff
@@ -20,9 +20,10 @@ Development takes place on GitHub, where the git-flow branch structure is used:
 
 ### Prepare development environment
 It is recommended to use Unix systems. You can use free Oracle Virtualbox and Ubuntu.  
-Setup script for Ubuntu system (python version might be 3.6+):
+Setup script for Ubuntu system (python version might be 3.7+):
```
Comment (briantist) [inline docs/CONTRIBUTE.md:23]: I realize it was like this in the original but if we're changing the line, as well update

```suggestion
Setup script for Ubuntu system (python version must be 3.7+):
```

**Review by briantist — APPROVED** [review]
not that my review counts toward the repo requirements, but thank you for this!

**Review by allburov — APPROVED** [review]
If it was a steak, it'd be well done!

**Pull request merged**